Source/diablo.cpp

```diff
@@ -746,29 +674,97 @@ LRESULT __stdcall GM_Game(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
 		PressChar(wParam);
 		return 0;
 	case WM_SYSKEYDOWN:
-		if (PressSysKey(wParam))
+		if(PressSysKey(wParam))
 			return 0;
-		return MainWndProc(hWnd, uMsg, wParam, lParam);
+		break;
 	case WM_SYSCOMMAND:
-		if (wParam == SC_CLOSE) {
+		if(wParam == SC_CLOSE) {
 			gbRunGame = FALSE;
 			gbRunGameResult = FALSE;
 			return 0;
 		}
-		return MainWndProc(hWnd, uMsg, wParam, lParam);
+		break;
+	case WM_MOUSEMOVE:
+		MouseX = LOWORD(lParam);
+		MouseY = HIWORD(lParam);
+		gmenu_on_mouse_move(LOWORD(lParam));
+		return 0;
+	case WM_LBUTTONDOWN:
+		MouseX = LOWORD(lParam);

```

Comment:
Will this still need further clean up to remove LOWORD and HIWORD or should they stay?

Reply:
Those are macros designed specifically for unpacking window message parameters.

Reply:
you could try something `(WORD)lParam` for LOWORD
and `lParam >> 16,` or `(WORD)(lParam>>16)` for HIWORD, if you prefer

or you could even do `WORD* temp = lParam;` and then do `temp[0]` for LOWORD and `temp[1]` for HIWORD (depending on endianess), if you prefer and it makes it more readable to you.

Reply:
That's what I was thinking, but if we are happy with LO/HIWORD then that's fine also, the other options are a bit messy anyway.